plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -143,9 +151,138 @@ func (k *KafkaConsumer) Init() error {
 	}
 
 	k.config = cfg
+
+	k.compileTopicRegexps()
+
+	if len(k.regexps) == 0 {
+		// There are no regexp-matched topics
+		k.allWantedTopics = k.Topics
+	}
+
 	return nil
 }
 
+func (k *KafkaConsumer) createTopicScanner() error {
+	if k.topicClient == nil {
+		client, err := sarama.NewClient(k.Brokers, k.config)
+		if err != nil {
+			return err
+		}
+		k.topicClient = client
+	}
+	return nil
+}
+
+func (k *KafkaConsumer) compileTopicRegexps() {
+	// While we can add new topics matching extant regexps, we can't
+	// update that list on the fly.  We compile them once at startup.
+	// Changing them is a configuration change and requires a restart.
+
+	if len(k.TopicRegexps) == 0 {
+		// Nothing to do.
+		return
+	}
+	k.regexps = make([]regexp.Regexp, 0, len(k.TopicRegexps))
+	// Let's optimistically assume that all of them compile
+	for _, r := range k.TopicRegexps {
+		re, err := regexp.Compile(r)
+		if err == nil {
+			k.regexps = append(k.regexps, *re)
+			k.Log.Infof("Added regular expression '%s' to topics", r)
+		} else {
+			k.Log.Errorf("Regular expression '%s' did not compile: '%w'; not adding to topics", r, err)
```

Comment:
You should return an error if compiling the regexp fails. This is a configuration error and should not be accepted.
```suggestion
			return fmt.Errorf("regular expression %q did not compile: '%w", r, err)
```